WcaOnRails/spec/features/edit_user_spec.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+RSpec.feature "Edit user" do
+  let(:admin) { FactoryBot.create(:admin) }
+  let(:existing_user) { FactoryBot.create(:user_with_wca_id) }
+  let(:new_person) { FactoryBot.create(:person) }
+  let(:new_user) { FactoryBot.create(:user) }
+
+  def navigate_to_form(user)
+    visit edit_user_path(user)
+  end
+
+  scenario "entering wca id", js: true do
+    sign_in admin
+    navigate_to_form(new_user)
+
+    # Entering an existing wca id
+    fill_in "WCA ID", with: existing_user.wca_id
+    find('input[type="submit"]').click
+
+    expect(page).to have_text I18n.t('users.errors.unique',
+                                     used_name: existing_user.name,
+                                     used_email: existing_user.email)
+
+    # Entering a valid wca id
+    fill_in "WCA ID", with: new_person.wca_id
+    find('input[type="submit"]').click
```

Comment:
I remember you had this before as a separate `submit_form` method, maybe it would be nice to do that again?

Reply:
No idea why would I would remove that. I have added it once again. :)